plugins/flatpak/gs-flatpak.c

```diff
@@ -1524,6 +1533,48 @@ gs_flatpak_add_updates (GsFlatpak *self, GsAppList *list,
 	return TRUE;
 }
 
+static gboolean
+is_endless_app (GsApp *app)
+{
+	return g_str_has_prefix (gs_app_get_id (app), "com.endlessm.");
+}
+
+static gboolean
+is_endless_content_runtime (GsApp *app)
+{
+	const gchar *app_id = gs_app_get_id (app);
+	return gs_app_get_kind (app) == AS_APP_KIND_RUNTIME &&
+		is_endless_app (app) && g_str_has_suffix (app_id, "Content");
+}
+
+static UpdatePriority
+get_app_update_priority (GsApp *app)
+{
+	if (gs_app_get_kind (app) == AS_APP_KIND_RUNTIME) {
+		if (is_endless_content_runtime (app))
+			return UPDATE_PRIORITY_HIGHEST;
+		return UPDATE_PRIORITY_LOW;
+	}
```

Comment:
Nitpick: You could reduce the nesting here, since `is_endless_content_runtime()` checks whether the `GsApp` is a runtime. So you could do:
```
if (is_endless_content_runtime (app)
  return UPDATE_PRIORITY_HIGHEST;
else if (gs_app_get_kind (app) == AS_APP_KIND_RUNTIME)
  return UPDATE_PRIORITY_LOW;
else if (is_endless_app (app))
  /* we prioritise Endless apps … */
  return UPDATE_PRIORITY_HIGH;
else
  return UPDATE_PRIORITY_NORMAL;
```

I think that would make the ordering of priorities a little more obvious.